blog/2023-02-16-do-you-need-to-build-auth/index.md

```diff
@@ -0,0 +1,124 @@
+---
+slug: do-you-need-to-build-auth
+authors: gao
+tags: [auth, dev, authentication, authorization]
+---
+
+# Do you need to build your own auth for apps?
+
+:::note Prologue
+Although article is inspired from personal dev experience and I tried my best to keep things objective, it's still non-negligible that I'm one of the creators of Logto (an auth product).
+
+I would like to point out this first for a better reading experience.
+:::
+
+I’ve seen a lot of developers are asking questions like “should I build my own auth for my app”. While the answer cannot be simple yes or no, I’d like to write an article to breakdown the implementation and demonstrate the pros and cons to help you decide.
+
+> **TL;DR** If you're still learning, auth will be a good challenge. If you want full control, it's worth building your own auth system. Otherwise, it depends on several conditions.
+
+## Intro
+
+As a developer, I have built many applications during my career. No matter the programming language, there is a common foundation I always need to construct: user auth.
```

Comment:
```suggestion
As a developer, I have built many applications during my career. Regardless of the programming language, there is a common foundation I always need to construct: user auth.
```

Reply:
they have the same meaning?

Reply:
I may be wrong, but if I use `no matter` in this case, I would say `No matter what programming language is...`

Reply:
I may be wrong, but if I use `no matter` in this case, I would say `No matter what programming language is...`

Reply:
oh right, AI is not good